drivers/gpio/gpio_stm32.c

```diff
@@ -292,18 +292,28 @@ static int gpio_stm32_clock_request(const struct device *dev, bool on)
 	return ret;
 }
 
+
 static inline uint32_t gpio_stm32_pin_to_exti_line(int pin)
 {
 #if defined(CONFIG_SOC_SERIES_STM32L0X) || \
 	defined(CONFIG_SOC_SERIES_STM32F0X)
 	return ((pin % 4 * 4) << 16) | (pin / 4);
 #elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32g0_exti)
 	return ((pin & 0x3) << (16 + 3)) | (pin >> 2);
+#elif DT_HAS_COMPAT_STATUS_OKAY(st_stm32h7rs_exti)
+	/*
+	 * No need to translate pin to a LL_SBS_EXTI_LINEn value
+	 * as the gpio_stm32_set_exti_source does not call LL_SBS_SetEXTISource
+	 * but uses HAL formula : keep pin value [0..15]
+	 */
+	return (uint32_t)pin;
```

Comment:
This operation is not in line anymore with the function name.
If changes have to be done in the driver implementation, it should be applied to the other series as well.

Reply:
changed due to   LL_SBS_SetEXTISource